src/dealer/DealerWallet.ts

```diff
@@ -0,0 +1,515 @@
+import _ from "lodash"
+import { yamlConfig } from "../config"
+import { accountDealerFtxPath, liabilitiesDealerFtxPath } from "../ledger/ledger"
+import { MainBook } from "../mongodb"
+import { OnChainMixin } from "../OnChain"
+import { ILightningWalletUser } from "../types"
+import { btc2sat, sleep } from "../utils"
+import { baseLogger } from "../logger"
+import { UserWallet } from "../userWallet"
+import assert from "assert"
+
+import { GenericExchange, SupportedExchanges, ApiConfig } from "./GenericExchange"
+
+const activeExchangeId: SupportedExchanges = "ftx"
+
+const key = process.env[`${activeExchangeId.toUpperCase()}_KEY`]
+const secret = process.env[`${activeExchangeId.toUpperCase()}_SECRET`]
+const password = process.env[`${activeExchangeId.toUpperCase()}_PASSWORD`]
+
+if (!key || !secret || !password) {
+  throw new Error(`Missing ${activeExchangeId} exchange environment variables`)
+}
+
+const activeApiConfig = new ApiConfig(activeExchangeId, key, secret, password)
+
+// TODO: move to the yaml config
+const simulateOnly = true
+
+export type IBuyOrSell = "sell" | "buy" | null
+
+// FtxDealer is a user of the wallet, therefore we are inheriting from UserWallet
```

Comment:
comment probably needs to be updated

Reply:
Done! If using LN with the exchange as we propose to do with OKEx, should the DealerWallet inherit from the LightningMixin?